processor/processor.go

```diff
@@ -593,78 +612,71 @@ func (proc *HandleT) processJobsForDest(jobList []*jobsdb.JobT, parsedEventList
 	proc.marshalSingularEvents.Start()
 	for idx, batchEvent := range jobList {
 
-		var eventList []interface{}
+		var singularEvents []types.SingularEventT
 		var ok bool
 		if parsedEventList == nil {
-			eventList, ok = misc.ParseRudderEventBatch(batchEvent.EventPayload)
+			singularEvents, ok = misc.ParseRudderEventBatch(batchEvent.EventPayload)
 		} else {
-			eventList = parsedEventList[idx]
-			ok = (eventList != nil)
+			singularEvents = parsedEventList[idx]
+			ok = (singularEvents != nil)
 		}
 		writeKey := gjson.Get(string(batchEvent.EventPayload), "writeKey").Str
 		requestIP := gjson.Get(string(batchEvent.EventPayload), "requestIP").Str
 		receivedAt := gjson.Get(string(batchEvent.EventPayload), "receivedAt").Time()
 
 		if ok {
 			//Iterate through all the events in the batch
-			for _, singularEvent := range eventList {
+			for _, singularEvent := range singularEvents {
 				//We count this as one, not destination specific ones
 				totalEvents++
 				//Getting all the destinations which are enabled for this
 				//event
-				destTypesFromConfig := getEnabledDestinationTypes(writeKey)
-				destTypes := integrations.GetDestinationIDs(singularEvent, destTypesFromConfig)
+				backendEnabledDestTypes := getBackendEnabledDestinationTypes(writeKey)
+				enabledDestTypes := integrations.FilterClientIntegrations(singularEvent, backendEnabledDestTypes)
 
-				// logger.Debug("=== destTypes ===", destTypes)
-				if len(destTypes) == 0 {
+				// logger.Debug("=== enabledDestTypes ===", enabledDestTypes)
+				if len(enabledDestTypes) == 0 {
 					logger.Debug("No enabled destinations")
 					continue
 				}
 				enabledDestinationsMap := map[string][]backendconfig.DestinationT{}
-				for _, destType := range destTypes {
+				for _, destType := range enabledDestTypes {
 					var enabledDestinationsList []backendconfig.DestinationT
 					if isReplayServer {
 						enabledDestinationsList = getReplayEnabledDestinations(writeKey, destType)
 					} else {
 						enabledDestinationsList = getEnabledDestinations(writeKey, destType)
 					}
 					enabledDestinationsMap[destType] = enabledDestinationsList
+
 					// Adding a singular event multiple times if there are multiple destinations of same type
-					if len(destTypes) == 0 {
```

Comment:
Instead of removing this, it seems like len(destTypes) should instead be changed to len(enabledDestinationsList)